ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,266 @@
+ */
+public struct DrawerModal {
+
+    public struct DrawerShimView: View {
+
+        private var drawer: Drawer<DrawerContentViewController>
+
+        public var drawerState: DrawerState {
+            return drawer.state
+        }
+
+        init(contentViewController: UIViewController) {
+            drawer = Drawer(content: DrawerContentViewController(contentViewController: contentViewController))
+        }
+
+        public var body: some View {
+            ZStack {
+                drawer.edgesIgnoringSafeArea(.all)
+            }
+        }
+    }
+
+    public struct DrawerContentViewController: UIViewControllerRepresentable {
+
+        private var contentView: UIViewController
+
+        init(contentViewController: UIViewController) {
+            self.contentView = contentViewController
+        }
+
+        public func makeUIViewController(context: Context) -> UIViewController {
+            return contentView
+        }
+
+        public func updateUIViewController(_ uiViewController: UIViewController, context: Context) {}
+
+        public typealias UIViewControllerType = UIViewController
+    }
+}
+
+@objc(MSFDrawerVnext)
+
+/// UIKit wrapper that exposes the SwiftUI Drawer implementation
+open class DrawerVnext: UIHostingController<DrawerModal.DrawerShimView> {
+
+    @objc open var drawerState: DrawerState {
+        return self.rootView.drawerState
+    }
+
+    @objc public init(contentViewController: UIViewController) {
+        super.init(rootView: DrawerModal.DrawerShimView(contentViewController: contentViewController))
+        transitioningDelegate = self
+        drawerState.delegate = self
+        view.backgroundColor = .clear
+        modalPresentationStyle = .overFullScreen
+    }
+
+    @objc required dynamic public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+    }
+}
+
+extension DrawerVnext: DrawerStateDelegate {
+    // two way binding if state is changed
+    func drawerDidCollapsed() {
+        dismiss(animated: true)
+    }
+}
+
+extension DrawerVnext: UIViewControllerTransitioningDelegate, UIViewControllerAnimatedTransitioning {
+    public func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+        return 0
+    }
+
+    public func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+
+        let fromView = transitionContext.viewController(forKey: .from)!.view!
+        let toView = transitionContext.viewController(forKey: .to)!.view!
+
+        let isPresentingDrawer = toView == view
+
+        let drawerView = isPresentingDrawer ? toView : fromView
+
+        // delegate animation to swiftui by changing state
+        if isPresentingDrawer {
+            transitionContext.containerView.addSubview(drawerView)
+            drawerView.frame = UIScreen.main.bounds
+            // Added a static interval as currently swiftui doesn't have an animation completion callback
```

Comment:
Do you think that the withAnimation function could be useful to avoid this?

https://developer.apple.com/documentation/swiftui/withanimation(_:_:)

Reply:
By nesting animation modifiers I believe you can eventually disable the animation of repositioning the drawer while animating just its presentation and avoiding the bug on the video below:


https://user-images.githubusercontent.com/68076145/103310566-c314ac00-49cc-11eb-9a00-b1c2af5c6643.mov

Reply:
This should be now working as expected